fe/src/main/java/org/apache/doris/planner/SingleNodePlanner.java

```diff
@@ -593,8 +595,20 @@ private void turnOffPreAgg(AggregateInfo aggInfo, SelectStmt selectStmt, Analyze
             olapNode.setIsPreAggregation(true, null);
         } while (false);
 
-        if ((root instanceof OlapScanNode) && turnOffReason != null) {
-            ((OlapScanNode) root).setIsPreAggregation(false, turnOffReason);
+        if ((root instanceof OlapScanNode)) {
+            OlapScanNode node = (OlapScanNode) root;
+            for (TableRef tableRef : selectStmt.getTableRefs()) {// Force Open PreAgg
```

Comment:
what do this logic means to do?

Reply:
We must have to match the table name，so we can set OpenPreAgg flags